### DISCUSSION TITLE
0.7.0 - Gapless and more

### DISCUSSION THREAD

**Discussion by basharovV** [basharovV/musicat#9] [Announcements]
_BREAKING RELEASE_. The way albums are stored internally has changed, please delete and re-import your library.
-   **Gapless playback**. The audio playback component has been re-built to support gapless playback (between files of the same sample rate), resampling (automatically switches on when the audio device doesn't support the file sample rate), and better performance than the previous web-based version.
-   **New: Queue**. Press "Q" to open up the new queue panel. You can drag and drop tracks from the library, re-arrange tracks in the queue, and quickly create a new playlist from the current queue. Multi-select works via Shift/Cmd + click. The queue has two modes: Custom and Same as Library. When in "Same as library" mode, playing a track from the library will reset the queue to library order.
<img width="864" alt="image" src="https://github.com/basharovV/musicat/assets/8410950/7e90e8fa-6d8f-47ec-8f0f-1c6d56c847a0">

-   **Waveform View**. Click on the wave icon next to the volume to open up the Waveform view. Here you can drag to create loop regions, or Cmd/Ctrl + Click to set markers of your favourite song moments. You can scroll to zoom in and pan around the waveform.

<img width="1615" alt="image" src="https://github.com/basharovV/musicat/assets/8410950/4b3f91fb-fc52-4935-a10e-32737f14d0af">

-   **Metadata fixes**. Parsing and writing metadata should be more stable now, as well as album artwork scanning for the Album grid.
-   **Library columns state is saved**. The order of columns, and which columns are visible is now remembered between app restarts.
-   **Lots of many things**. Revised UI panel layout. Fuzzier search. New release notes. Remember window size and position. Smoother oscilloscope. Folder watcher now detects new folder additions.

<hr /><em>This discussion was created from the release <a href='https://github.com/basharovV/musicat/releases/tag/v0.7.0'>0.7.0 - Gapless and more</a>.</em>

**Comment by jmccarthy86** [comment]
I love the waveform view! I've not tried the app yet, just found you now whilst researching tauri and audio. I'm playing around with node/c backends with web frontends. Early days. 

This app is inspiring, it's taking up where WinAmp left off 😅👌👌

**Reply by basharovV** [reply]
Thank you! I'm just playing around as well, I use this player on a daily basis but it's also a great playground to try some funky features. And I'm quite happy with the Rust backend and Svelte frontend approach. 

I just released version 0.8 which has a new Internet Archive view, check it out :)
